sdks/java/core/src/main/java/org/apache/beam/sdk/testing/TestPipeline.java

```diff
@@ -477,6 +478,12 @@ public static PipelineOptions testingPipelineOptions() {
     }
   }
 
+  public PipelineRunner toPipelineRunner() {
```

Comment:
This method feels very out of place. I would instead simply ignore the provided pipeline in your testRunPTransform and create a new one from scratch.

Reply:
Ok, I removed the method. Also, I moved the test case to org.apache.beam.sdk.runners.PipelineRunnerTest, as it no longer has dependencies on objects in PipelineTest.